src/components/modules/api/blocks.ts

```diff
@@ -247,6 +249,27 @@ export default class BlocksAPI extends Module {
     return new BlockAPI(insertedBlock);
   }
 
+  /**
+   * Retrieves default block data by creating fake block.
+   * Merges retrieved data with specified data object.
+   *
+   * @param toolName - block tool name
+   * @param dataOverrides - object containing overrides for default block data
+   */
+  public composeBlockData = async (toolName: string, dataOverrides = {}): Promise<BlockToolData> => {
+    const tool = this.Editor.Tools.blockTools.get(toolName);
+    const block = new Block({
+      tool,
+      api: this.Editor.API,
+      readOnly: true,
+      data: {},
+      tunesData: {},
+    });
+    const blockData = await block.data;
+
+    return Object.assign(blockData, dataOverrides);
```

Comment:
I think, merging is too specific thing. So it should be done outside ot this method, at the `toolbox.ts@insertNewBlock`. 

You've added the new public API method, so it should be useful for developers. I can't imagine any usecases for now, but let it be there. 

Also, please add a changelog about this new method.

Reply:
done